validation-test/compiler_crashers_2_fixed/0163-sr8033.swift

```diff
@@ -7,4 +7,4 @@ protocol P1 {
 }
 extension Foo: P1 where A : P1 {} // expected-error {{unsupported recursion for reference to associated type 'A' of type 'Foo<T>'}}
 // expected-error@-1 {{type 'Foo<T>' does not conform to protocol 'P1'}}
-// expected-error@-2 {{type 'Foo<T>' in conformance requirement does not refer to a generic parameter or associated type}}
+// expected-error@-2 {{type '<<error type>>' in conformance requirement does not refer to a generic parameter or associated type}}
```

Comment:
Why did this change?

Reply:
Looks like the result of a `!ty` -> `ty->hasError()` transition somewhere in `TypeCheckType.cpp`. The error message is bogus though, anyway..

Reply:
The second time we come around to add these requirements, the base is invalidated somewhere else in the stack. I think we should just not emit this diagnostic if the subject is an `ErrorType` - it doesn't make sense here in the first place.